flocker/node/agents/ebs.py

```diff
@@ -107,6 +115,63 @@ def _wait_for_volume(expected_volume,
             )
 
 
+def _check_blockdevice_size(device, size):
+    """
+    Helper function to check if the size of block device with given
+    suffix matches the input size.
+
+    :param unicode device: Name of the block device to check for size.
+    :param int size: Size, in SI metric bytes, of device we are interested in.
+
+    :returns: True if a block device with given name has given size.
+        False otherwise.
+    """
+    device_name = b"/dev/" + device.encode("ascii")
+
+    # Retrieve size of device as OS sees it using `lsblk`.
+    # Requires util-linux-ng package on CentOS, and
+    # util-linux on Ubuntu.
+    # Required package is installed by default
+    # on Ubuntu 14.04 and CentOS 7.
+    command = [b"/bin/lsblk", b"-nb", b"-o", b"SIZE", device_name]
+
+    # Get the base device size, which is the first line in
+    # `lsblk` output. Ignore partition sizes.
+    command_output = check_output(command).split(b'\n')[0]
+    device_size = int(command_output.strip().decode("ascii"))
+
+    return size == device_size
+
+
+def _wait_for_new_device(base, size, time_limit=60):
+    """
+    Helper function to wait for up to 60s for new
+    EBS block device (`/dev/sd*` or `/dev/xvd*`) to
+    manifest in the OS.
+
+    :param list base: List of baseline block devices
+        that existed before execution of operation that expects
+        to create a new block device.
+    :param int size: Size of the block device we are expected
+        to manifest in the OS.
+
+    :returns: formatted string name of the new block device.
+    :rtype: unicode
+    """
+    start_time = time.time()
+    elapsed_time = time.time() - start_time
+    while elapsed_time < time_limit:
+        for device in list(set(FilePath(b"/sys/block").children()) -
+                           set(base)):
+            device_name = FilePath.basename(device)
+            if (device_name.startswith((b"sd", b"xvd")) and
+                    _check_blockdevice_size(device_name, size)):
+                new_device = u'/dev/' + device_name
```

Comment:
`device_name` is `bytes` (since it came from `FilePath.basename`) - decode it as `ascii` here, probably.